File: src/player.hs
{-# LANGUAGE DeriveGeneric #-}

module Player (Player(White, Black), next) where

import Data.Aeson
import GHC.Generics

-- Player & Related Functions--
data Player = White | Black 
	deriving (Eq, Show, Read, Ord, Bounded, Enum, Generic)

instance FromJSON Player
instance ToJSON   Player


next :: Player -> Player
next White = Black
next Black = White

File: frontend/src/ts/abalone/player.ts
module Abalone {
	export enum Player {White, Black}
	export function next(p: Player) {
		return (p === Player.White) ? Player.Black : Player.White;
	}

	export function player2str(p: Player): string {
		return p === Player.White ? "White" : "Black"
	}
	export function str2player(s: String): Player {
		switch (s) {
			case "White":
			return Player.White;
			case "Black": 
			return Player.Black;
			default:
			return null;
		}
	}
}